test/unit/math/rev/core/var_test.cpp

```diff
@@ -13,62 +14,51 @@ struct AgradRev : public testing::Test {
   }
 };
 
-TEST_F(AgradRev, ctorOverloads) {
-  using stan::math::var;
-  using stan::math::vari;
-
+template <typename T, typename S>
+void ctor_overloads_impl() {
+  using stan::math::var_value;
+  using stan::math::vari_value;
+  using stan::math::test::type_name;
+  // standard constructor
+  EXPECT_FLOAT_EQ(3.7, var_value<T>(3.7).val())
+      << "Failed For T: " << type_name<T>() << "\n";
   // make sure copy ctor is used rather than casting vari* to unsigned int
-  EXPECT_FLOAT_EQ(12.3, var(new vari(12.3)).val());
-
-  // double
-  EXPECT_FLOAT_EQ(3.7, var(3.7).val());
-
-  // long double
-  EXPECT_FLOAT_EQ(3.7, var(static_cast<long double>(3.7)).val());
-
-  // float
-  EXPECT_FLOAT_EQ(3.7, var(static_cast<float>(3.7)).val());
-
-  // bool
-  EXPECT_FLOAT_EQ(1, var(static_cast<bool>(true)).val());
-
-  // char
-  EXPECT_FLOAT_EQ(3, var(static_cast<char>(3)).val());
-
-  // short
-  EXPECT_FLOAT_EQ(1, var(static_cast<int16_t>(1)).val());
-
-  // int
-  EXPECT_FLOAT_EQ(37, var(static_cast<int>(37)).val());
-
-  // long
-  EXPECT_FLOAT_EQ(37, var(static_cast<int32_t>(37)).val());
-
-  // unsigned char
-  EXPECT_FLOAT_EQ(37, var(static_cast<unsigned char>(37)).val());
-
-  // unsigned short
-  EXPECT_FLOAT_EQ(37, var(static_cast<uint16_t>(37)).val());
-
-  // unsigned int
-  EXPECT_FLOAT_EQ(37, var(static_cast<unsigned int>(37)).val());
-
-  // unsigned int (test conflict with null pointer)
-  EXPECT_FLOAT_EQ(0, var(static_cast<unsigned int>(0)).val());
-
-  // unsigned long
-  EXPECT_FLOAT_EQ(37, var(static_cast<uint32_t>(37)).val());
-
-  // unsigned long (test for conflict with pointer)
-  EXPECT_FLOAT_EQ(0, var(static_cast<uint32_t>(0)).val());
+  EXPECT_FLOAT_EQ(12.3, var_value<T>(new vari_value<T>(12.3)).val())
```

Comment:
It's weird to make `vari_value` s out of ints. This might break in the future but I guess the test is fine now.

Reply:
Ignore my comment. That isn't happening here.